### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by shhavel**
- use each_with_object instead each;
- use define_singleton_method instead method_missing.

### CODE REVIEW

File: lib/raph/parser/base_parser.rb [hunk 1]
```diff
@@ -17,33 +17,20 @@ class BaseParser
...
-          downcase.
-          to_sym
-      end
-
-      private
-
-      # Returns name of current class.
-      def class_name
-        self.class.name.split('::').last || ''
+        str.sub(/^-+/, '').gsub(/([A-Z\d]+)([A-Z][a-z])/, '\1_\2').gsub(/([a-z\d])([A-Z])/,'\1_\2').tr('-', '_').downcase.to_sym
```
Comment (veelenga) [inline lib/raph/parser/base_parser.rb]: This line becoming a mess. Can we use [multi-line chaining with a leading dot](https://github.com/bbatsov/ruby-style-guide#user-content-consistent-multi-line-chains) here ? It just a matter of style.

**Comment by veelenga** [comment]
Your refactoring is great! I have appreciated `each_with_object` and `define_singleton_method`. 
Thanks!

File: lib/raph/parser/base_parser.rb (hunk 1 above)
Comment (shhavel) [inline lib/raph/parser/base_parser.rb]: Sure, in general it's a matter of style, probably will be clearer to split.  
Thanks.

**Comment by veelenga** [comment]
Thank you :+1:

**Pull request merged**